WidgetExtension/Variants/Hashtag/HashtagWidgetView.swift

```diff
@@ -89,55 +91,57 @@ struct HashtagWidgetView: View {
 
-        let textColor = UIColor(named: "Colors/TextColor")
-        let accentColor = UIColor(named: "Colors/AccentColor")
-        let font = UIFontMetrics(forTextStyle: .headline).scaledFont(for: .systemFont(ofSize: fontSize, weight: fontWeight))
+        let textColor = UIColor(named: "Colors/TextColor") ?? .label
+        let accentColor = UIColor(named: "Colors/AccentColor") ?? .red
+        let font = UIFontMetrics(forTextStyle: .body).scaledFont(for: .systemFont(ofSize: fontSize, weight: fontWeight))
 
         let attributedString: AttributedString
-
         // 1. Render status-content as HTML ...
-        if let data = htmlString.data(using: .utf8),
+        if let data = statusContent.data(using: .utf8),
            let renderedString = try? NSAttributedString(data: data,
                                                         options: [
                                                             .documentType: NSAttributedString.DocumentType.html,
                                                             .characterEncoding: NSUTF8StringEncoding
                                                         ],
                                                         documentAttributes: nil) {
 
-
-            // 2. get the raw string ...
-            let rawString = renderedString.string
-
-            // 3. ... and use regex to get the hashtags
-            let hashtagRegex = try? NSRegularExpression(pattern: "(#\\w*)")
-            let hashtagRanges = hashtagRegex?.matches(
-                in: rawString,
-                range: NSMakeRange(0, rawString.count)
-            ).compactMap { NSMakeRange($0.range.location, $0.range.length) } ?? []
-
-            let mutableAttributedString = NSMutableAttributedString(
-                string: rawString,
-                attributes: [
-                    .foregroundColor: textColor ?? UIColor.label,
-                    .font: font
-                ]
-            )
-
-            // 4. color the hashtags
-            hashtagRanges.forEach {
-                mutableAttributedString.setAttributes([
-                    .foregroundColor: accentColor ?? UIColor.red
-                ], range: $0)
-            }
-
-            attributedString = AttributedString(mutableAttributedString)
+            // 2. let MetaTextKit do the rest
+            let content = MastodonContent(content: renderedString.string, emojis: [:])
+            attributedString = MastodonMetaContent.convert(text: content)
+                .attributedString(textColor: textColor, accentColor: accentColor, font: font)
         } else {
             // this is a fallback
-            attributedString = AttributedString(NSAttributedString(string: htmlString))
+            attributedString = AttributedString(NSAttributedString(string: statusContent))
         }
 
         self.init(attributedString)
     }
 }
+
+#warning("Replace this once we update MetaTextKit to `4.5.2`. We should do that.")
```

Comment:
MetaTextKit 3 and above use the new TextKit 2 APIs, and it seems like TextKit 2 has a bunch of bugs / missing functionality so @MainasuK didn’t recommend updating yet.

Reply:
Ah, I wasn't aware of that, again: Thank you :)

Reply:
Yes. Although I fix some bugs like the [wrong RTL layout](https://github.com/TwidereProject/MetaTextKit/commit/c2a3d5e4355c1e3c9046c89433b28795f3374876) when layout using TextKit 2 in the latest MetaTextKit (v4). But the paragraph spacing still does not be respected by the text engine. 

I recommend using the native UITextView with NSAttributedString which export from the parser. Here is an [example](https://github.com/TwidereProject/TwidereX-iOS/blob/develop/TwidereSDK/Sources/TwidereUI/UIViewRepresentable/TextViewRepresentable.swift) I'm working on. And this requires iOS 16.0+. So staying in the **stable** old version and upgrading to this in the future is a good idea. :)

Reply:
Unfortunately we can't use UIKit-stuff in Widgets. It's just not possible :) So we'll just wait then? :D

Reply:
@zeitschlag Oh. I think the custom emoji will not display if the UIKit is not usable in the Widget. Due to AttributedString doesn't support custom text attachments. Maybe it's acceptable.